api/src/wfl/once.clj

```diff
@@ -1,19 +1,43 @@
 (ns wfl.once
-  "Manage credentials."
+  "Manage credentials and configurations."
   (:require [clojure.data.json :as json]
             [clojure.java.io   :as io]
             [clojure.string    :as str]
+            [clojure.tools.logging :as log]
             [wfl.environments :as env]
+            [vault.client.http]         ; vault.core needs this
+            [vault.core :as vault]
             [wfl.util         :as util])
   (:import [com.google.auth.oauth2 GoogleCredentials]))
 
+;; Only load the relevant environment variables
+;;
 (defonce the-system-environments
-  (reduce #(assoc %1 %2 (System/getenv %2)) {}
-          ["COOKIE_SECRET"
-           "CROMWELL"
-           "GOOGLE_APPLICATION_CREDENTIALS"
-           "WFL_OAUTH2_CLIENT_ID"
-           "WFL_DEPLOY_ENVIRONMENT"]))
+  (delay (reduce #(assoc %1 %2 (System/getenv %2)) {}
```

Comment:
`memoize` looks exactly like what I was looking for, TIL thanks!

> Maybe rename "environments.clj" to "environment.clj" and export the function get?
So we are going to scope `environment.clj` module to be the "envrionment variables" module from now on? I think that's a good idea! It may also make sense to rename `once` to something like `auth.clj`

Reply:
Maybe. I defer to you if you think its a good idea or not!
`auth` does seem like a better name than `once` for what it now contains.

Reply:
> rename `once` to something like `auth.clj`

Yes there is no `defoncé` left.